src/include/expression/parser_expression.h

```diff
@@ -26,54 +26,58 @@ namespace expression {
 class ParserExpression : public AbstractExpression {
  public:
   ParserExpression(ExpressionType type, char* name_)
- : AbstractExpression(type) {
+      : AbstractExpression(type) {
     name = name_;
   }
 
-  ParserExpression(ExpressionType type, char* name_, char* column_)
-  : AbstractExpression(type) {
+  ParserExpression(ExpressionType type, char* name_, char* column_or_database_)
+      : AbstractExpression(type) {
     name = name_;
-    column = column_;
+    if (type == peloton::EXPRESSION_TYPE_COLUMN_REF) {
+      column = column_or_database_;
+    } else if (type == peloton::EXPRESSION_TYPE_TABLE_REF) {
+      database = column_or_database_;
+    }
```

Comment:
I can understand the logic here after reading other parts of the code, but is it possible to separate this to two different functions? This might lead to confusion.